rust/feature-flags/src/flags/flag_request.rs

```diff
@@ -96,6 +100,20 @@ impl FlagRequest {
         }
     }
 
+    /// Sanitizes JSON by replacing unquoted NaN/Infinity with null, preserving quoted strings
+    /// This matches Python decide endpoint behavior: parse_constant=lambda x: None
+    fn sanitize_unquoted_non_finite_values(json_str: &str) -> String {
+        static NEG_INF_RE: Lazy<Regex> =
+            Lazy::new(|| Regex::new(r"(?i)-infinity(\s*[,}\]])").unwrap());
```

Comment:
Don't we need a word boundary before the `-Infinity` just like we do before line 109 where we handle `Infinity`?

```suggestion
            Lazy::new(|| Regex::new(r"(?i)\b-infinity(\s*[,}\]])").unwrap());
```